src/app/app-routing.module.ts

```diff
@@ -2,18 +2,18 @@ import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
 import { HomeComponent } from './home/home.component';
-import Auth from './_utils/Auth';
 import { AdminAPIComponent } from './admin-api/admin-api.component';
+import { Auth, PermissionGuard } from "./_utils"
 
 const routes: Routes = [
   { path: 'admin', component: AdminAPIComponent, canActivate: [Auth], },
-  { path: 'admin/:id', component: AdminAPIComponent, canActivate: [Auth] },
+  { path: 'admin/:id', component: AdminAPIComponent, canActivate: [Auth, PermissionGuard] },
   { path: '**', component: HomeComponent },
 ];
 
 @NgModule({
   imports: [RouterModule.forRoot(routes)],
   exports: [RouterModule],
-  providers: [Auth]
+  providers: [Auth, PermissionGuard]
```

Comment:
when it comes to a list of providers, may I ask you to read a bit about it, 
how it will be provided, what injector will have Permission Guard, considering you have it in a list of providers here and you also have `providedIn: root` param in decorator

https://angular.io/guide/providers

video: https://www.youtube.com/watch?v=G8zXugcYd7o
https://indepth.dev/posts/1090/angulars-root-and-any-provider-scopes